ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/ProfitLossPercentageCriterion.java

```diff
@@ -0,0 +1,46 @@
+package org.ta4j.core.analysis.criteria;
+
+import org.ta4j.core.TimeSeries;
+import org.ta4j.core.Trade;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.num.Num;
+
+/**
+ * Profit and loss in percentage criterion.
+ * </p>
+ * The profit or loss in percentage over the provided {@link TimeSeries series}.
+ * https://www.investopedia.com/ask/answers/how-do-you-calculate-percentage-gain-or-loss-investment/
+ */
+public class ProfitLossPercentageCriterion extends AbstractAnalysisCriterion {
+
+    @Override
+    public Num calculate(TimeSeries series, TradingRecord tradingRecord) {
+        return tradingRecord.getTrades().stream()
+                .filter(trade -> trade.isClosed())
+                .map(trade -> calculateProfitLossInPercentage(series, trade))
+                .reduce(series.numOf(0), (profit1, profit2) -> profit1.plus(profit2));
```

Comment:
Lambda can be replaced with method reference:
instead of '(profit1, profit2) -> profit1.plus(profit2)' you can simplify to: Num::plus

Reply:
skip